solver/src/driver.rs

```diff
@@ -63,6 +68,35 @@ impl Driver {
         }
     }
 
+    // Should this go here or on price_estimate.rs?
+    async fn collect_estimated_prices(
+        &self,
+        limit_orders: &[LimitOrder],
+    ) -> Result<HashMap<H160, BigRational>> {
+        if limit_orders.is_empty() {
+            return Ok(hashmap![]);
+        }
+        // Computes set of traded tokens (limit orders only).
+        let tokens: Vec<H160> = limit_orders
+            .iter()
+            .flat_map(|lo| vec![lo.sell_token, lo.buy_token].into_iter())
+            .sorted()
+            .dedup()
+            .collect();
+
+        // For ranking purposes it doesn't matter how the external price vector is scaled.
+        // If we use native_token here instead, should it come as another ctor parameter?
+        let denominator_token: H160 = tokens[0];
```

Comment:
I think it makes sense to use the native token here, as it will likely be the most liquid and therefore give the most accurate price estimations. Can be passed into the driver as ctor arg as you suggested.

Reply:
Yes, but notice that we don't care how accurate the actual price estimations are for the purposes of ranking. On the other hand, it might be better for logging/debugging. Done here e41c35.